### DISCUSSION TITLE
Web Executor with Chrome and Firefox

### DISCUSSION THREAD

**Discussion by dave-poe** [ovh/venom#515] [Q&A]
Hi folks. 

After great success with the grpc and api executor tests, our team have been looking into using the web executor for some light UI checks. Unfortunately we've been running into difficulties getting it working.

Just wondering if anyone had success or similar issues when using the web executor? 

Agouti is in a deprecated state and hasn't seen changes in a long time which I think may be contributing to the problems.

Attempting to use the chrome driver (Chrome 98) 
`failed to navigate: request unsuccessful: invalid session id`
Attempting to use gecko driver the page launches, but any attempted clicks result in the following (Firefox 97.0.1)
`failed to click on selection 'CSS: a[href*="blah"] [single]': request unsuccessful: HTTP method not allowed`

phantomjs is no longer maintained, so we are trying with chrome and firefox. I tried phantomjs out of interest and received the following on a click
`"name":"Invalid Command Method"`

Thanks in advance
Dave

**Comment by kevinramage** [comment]
Hello,

For the issue "failed to navigate: request unsuccessful: invalid session id", this issue linked to an issue to run the driver, often the issue is an invalid version between the browser driver and the driver.

You can find below chrome driver recommandations:
If you are using Chrome version 99, please download ChromeDriver 99.0.4844.35
If you are using Chrome version 98, please download ChromeDriver 98.0.4758.102
If you are using Chrome version 97, please download ChromeDriver 97.0.4692.71

Which version do you use for chrome driver and chrome ?

For the issue "request unsuccessful: HTTP method not allowed", I need more information about target web site to try to help you.
Can you provide an example of the target HTML page ?

Regards

**Reply by dave-poe** [reply]
Thanks for the reply @kevinramage . 

ChromeDriver version is 98.0.4758.102 and Chrome is 98.0.4758.109, so that seems to be good. 

Let me try to get a reproducible example with an external site. 

Dave

**Reply by dave-poe** [reply]
Hi @kevinramage 

If I use https://github.com/ovh/venom/blob/master/tests/web.yml as an example for chrome execution (commenting out the phantom part and uncommenting the chrome + args) I get the following errors

ChromeDriver version is 98.0.4758.102 and Chrome is 98.0.4758.109,

`TestCase-Get-URL-and-check-title FAILURE
Testcase "TestCase Get URL and check title", step #0: failed to navigate: request unsuccessful: invalid session id (./web.yml:19)
Testcase "TestCase Get URL and check title", step #1: Cannot count element input[name="custname"]: failed to select elements from selection 'CSS: input[name="custname"] [single]': request unsuccessful: invalid session id (./web.yml:27)
Testcase "TestCase Get URL and check title", step #2: Cannot find element input[name="custname"]: failed to select elements from selection 'CSS: input[name="custname"] [single]': request unsuccessful: invalid session id (./web.yml:33)
Testcase "TestCase Get URL and check title", step #3: Cannot count element button: failed to select elements from selection 'CSS: button [single]': request unsuccessful: invalid session id (./web.yml:39)`

**Comment by kevinramage** [comment]
Hello,

I check and I reproduce the issue, there are a communication issue between the web driver and the browser during the session creation "cannot parse args". I check venom code and due to a refactoring, web driver argument is now retrieve with the method "StringVarFromCtx" instead of using the method "StringSliceVarFromCtx".
I create the issue [518](https://github.com/ovh/venom/issues/518) to trace the issue.
I create a pull request to fix the issue: https://github.com/ovh/venom/pull/519

Regards,

**Reply by dave-poe** [reply]
Fantastic. Thanks @kevinramage
